diff -rauN bdftopcf/bdfint.h bdftopcf-tcc-patch/bdfint.h
--- bdftopcf/bdfint.h	2022-10-15 20:56:42.000000000 +0200
+++ bdftopcf-tcc-patch/bdfint.h	2024-10-06 15:38:33.773093095 +0200
@@ -65,8 +65,8 @@
     BOOL        haveDefaultCh;
 } bdfFileState;
 
-extern void bdfError(const char *message, ...) _X_ATTRIBUTE_PRINTF(1, 2);
-extern void bdfWarning(const char *message, ...) _X_ATTRIBUTE_PRINTF(1, 2);
+extern void bdfError(const char *message, ...);
+extern void bdfWarning(const char *message, ...);
 extern unsigned char *bdfGetLine(FontFilePtr file, unsigned char *buf, int len);
 extern Atom bdfForceMakeAtom(const char *str, int *size);
 extern Atom bdfGetPropertyValue(char *s);
diff -rauN bdftopcf/pcfwrite.c bdftopcf-tcc-patch/pcfwrite.c
--- bdftopcf/pcfwrite.c	2022-10-15 20:56:42.000000000 +0200
+++ bdftopcf-tcc-patch/pcfwrite.c	2024-10-06 15:40:23.699763542 +0200
@@ -45,7 +45,7 @@
 
 static CARD32 current_position;
 
-static void _X_ATTRIBUTE_PRINTF(1, 2)
+static void
 pcfError(const char *message, ...)
 {
     va_list args;
