diff -rauN dropbear-2024.85/src/default_options.h dropbear-2024.85-path-patch/src/default_options.h
--- dropbear-2024.85/src/default_options.h	2024-04-25 16:30:00.000000000 +0200
+++ dropbear-2024.85-path-patch/src/default_options.h	2024-06-15 17:19:13.154009646 +0200
@@ -311,7 +311,7 @@
 
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
-#define XAUTH_COMMAND "/usr/bin/xauth -q"
+#define XAUTH_COMMAND "/bin/xauth -q"
 
 
 /* If you want to enable running an sftp server (such as the one included with
@@ -324,7 +324,7 @@
 
 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
-#define DROPBEAR_PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#define DROPBEAR_PATH_SSH_PROGRAM "/bin/dbclient"
 
 /* Whether to log commands executed by a client. This only logs the
  * (single) command sent to the server, not what a user did in a
@@ -360,7 +360,7 @@
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
-#define DEFAULT_ROOT_PATH "/usr/sbin:/usr/bin:/sbin:/bin"
+#define DEFAULT_PATH "/bin"
+#define DEFAULT_ROOT_PATH "/bin"
 
 #endif /* DROPBEAR_DEFAULT_OPTIONS_H_ */
