diff -rauN nbd/nbd-client.c nbd-netlink-patch/nbd-client.c
--- nbd/nbd-client.c	2026-04-09 17:47:57.000000000 +0200
+++ nbd-netlink-patch/nbd-client.c	2026-04-10 09:51:48.804936229 +0200
@@ -1545,9 +1545,11 @@
 		netlink_configure(index, sockfds, flags, identifier);
 		
 		/* If persist mode is enabled, start the monitoring loop */
+#if HAVE_NETLINK
 		if (cur_client->persist_mode) {
 			return persist_mode_main(index, sockfds, flags);
 		}
+#endif
 		
 		return 0;
 	}
