diff -rauN a/config-extra.mak b/config-extra.mak
--- a/config-extra.mak	1970-01-01 01:00:00.000000000 +0100
+++ b/config-extra.mak	2025-07-17 15:44:55.758597384 +0200
@@ -0,0 +1,6 @@
+NATIVE_DEFINES += -march=i486 -mcpu=i486
+ROOT-i386 = @@BASE@@/build/stage1
+CRT-i386  = @@BASE@@/build/stage1/lib
+LIB-i386  = @@BASE@@/build/stage1/lib
+INC-i386  = @@BASE@@/build/stage1/include
+DEF-i386  += -D__linux__
diff -rauN a/configure b/configure
--- a/configure	2025-07-17 09:43:32.000000000 +0200
+++ b/configure	2025-07-17 15:44:19.539586512 +0200
@@ -440,7 +440,7 @@
 
 CONFTEST=./conftest$EXESUF
 if test -z "$cross_prefix" \
-    && $cc -o $CONFTEST "$source_path/conftest.c" \
+    && $cc -nostdlib -I@@BASE@@/build/stage0/include -o $CONFTEST @@BASE@@/build/stage0/lib/crt1.o @@BASE@@/src/stage1/tinycc-@@TINYCC_VERSION@@/conftest.c @@BASE@@/build/stage0/lib/libc.a @@BASE@@/build/stage0/lib/tcc/i386-libtcc1.a \
     && $CONFTEST 2>/dev/null; then
   cc_name="$($CONFTEST compiler)"
   gcc_major="$($CONFTEST version)"
diff -rauN a/Makefile b/Makefile
--- a/Makefile	2025-07-17 09:43:32.000000000 +0200
+++ b/Makefile	2025-07-17 15:51:28.016657573 +0200
@@ -36,13 +36,13 @@
  NATIVE_TARGET = $(ARCH)-win$(if $(findstring arm,$(ARCH)),ce,32)
 else
  CFG = -unx
- LIBS+=-lm
- ifneq ($(CONFIG_ldl),no)
-  LIBS+=-ldl
- endif
- ifneq ($(CONFIG_pthread),no)
-  LIBS+=-lpthread
- endif
+# LIBS+=-lm
+# ifneq ($(CONFIG_ldl),no)
+#  LIBS+=-ldl
+# endif
+# ifneq ($(CONFIG_pthread),no)
+#  LIBS+=-lpthread
+# endif
  # make libtcc as static or dynamic library?
  ifeq ($(CONFIG_static),no)
   LIBTCC=libtcc$(DLLSUF)
@@ -256,11 +256,12 @@
 	$S./c2str.exe $< $@
 
 c2str.exe : conftest.c
-	$S$(CC) -DC2STR $< -o $@
+	gcc -DC2STR $< -o $@
 
 # target specific object rule
 $(X)%.o : %.c $(LIBTCC_INC)
-	$S$(CC) -o $@ -c $< $(addsuffix ,$(DEFINES) $(CFLAGS))
+	$S$(CC) -o $@ -c $< $(addsuffix ,$(DEFINES) $(CFLAGS)) \
+	-I@@BASE@@/build/stage0/include -nostdlib
 
 # additional dependencies
 $(X)tcc.o : tcctools.c
@@ -268,7 +269,9 @@
 
 # Host Tiny C Compiler
 tcc$(EXESUF): tcc.o $(LIBTCC)
-	$S$(CC) -o $@ $^ $(addsuffix ,$(LIBS) $(LDFLAGS) $(LINK_LIBTCC))
+	$S$(CC) -o $@ $^ $(addsuffix ,$(LIBS) $(LDFLAGS) $(LINK_LIBTCC)) \
+	-nostdlib @@BASE@@/build/stage0/lib/crt1.o @@BASE@@/build/stage0/lib/libc.a \
+	@@BASE@@/build/stage0/lib/tcc/i386-libtcc1.a
 
 # Cross Tiny C Compilers
 # (the TCCDEFS_H dependency is only necessary for parallel makes,
@@ -281,7 +284,9 @@
 	@$(MAKE) --no-print-directory $@ CROSS_TARGET=$* ONE_SOURCE=$(or $(ONE_SOURCE),yes)
 
 $(CROSS_TARGET)-tcc$(EXESUF): $(TCC_FILES)
-	$S$(CC) -o $@ $^ $(LIBS) $(LDFLAGS)
+	$S$(CC) -o $@ $^ $(LIBS) $(LDFLAGS) \
+	-nostdlib @@BASE@@/build/stage0/lib/crt1.o @@BASE@@/build/stage0/lib/libc.a \
+	@@BASE@@/build/stage0/lib/tcc/i386-libtcc1.a
 
 # profiling version
 tcc_p$(EXESUF): $($T_FILES)
