diff -rauN tinycc/i386-asm.c tinycc-asm-hint-patches/i386-asm.c
--- tinycc/i386-asm.c	2026-01-22 12:48:40.000000000 +0100
+++ tinycc-asm-hint-patches/i386-asm.c	2026-01-22 13:05:42.585825009 +0100
@@ -1214,6 +1214,8 @@
         case 'i':
         case 'm':
         case 'g':
+        case 't':
+        case 'u':
             pr = 4;
             break;
         default:
@@ -1465,6 +1467,12 @@
                 }
             }
             break;
+        case 't':
+            // TODO "Top of 80387 floating-point stack (%st(0))"
+            break;
+        case 'u':
+            // TODO "Second from top of 80387 floating-point stack (%st(1))"
+            break;
         default:
             tcc_error("asm constraint %d ('%s') could not be satisfied",
                   j, op->constraint);
