diff -rauN tinyxlib/libXfont/bitmap/bdfread.c tinyxlib-tcc-patch/libXfont/bitmap/bdfread.c
--- tinyxlib/libXfont/bitmap/bdfread.c	2023-07-13 15:03:17.000000000 +0200
+++ tinyxlib-tcc-patch/libXfont/bitmap/bdfread.c	2023-07-27 16:34:24.360482336 +0200
@@ -54,6 +54,7 @@
 
 #ifndef FONTMODULE
 #include <ctype.h>
+#include <stdint.h>
 #endif
 #include "../include/fntfilst.h"
 #include "../include/fontutil.h"
