diff -rauN uflbbl/src/boot.asm uflbbl-boot-options/src/boot.asm
--- uflbbl/src/boot.asm	2023-07-13 08:47:23.000000000 +0200
+++ uflbbl-boot-options/src/boot.asm	2023-07-15 16:30:46.609197671 +0200
@@ -468,10 +468,10 @@
 	mov [READ_STATE], byte STATE_READ_INITRD
 ; qemu initrd start location 7fab000, 133869568 (this is 128MB) too high for us,
 ; kernel gives us alignment hints and hints where to load initrd to?
-; let's use 8MB, TODO: does the kernel release the initial ramdisk? I think so. is
+; let's use 6MB, TODO: does the kernel release the initial ramdisk? I think so. is
 ; it relocating it's structures? or do we get fragmented heap and stuff?
-	a32 mov [READ_DESTINATION_PTR], dword 0x00800000
-	a32 mov [INITRD_ADDRESS], dword 0x00800000
+	a32 mov [READ_DESTINATION_PTR], dword 0x00580000
+	a32 mov [INITRD_ADDRESS], dword 0x00580000
 	a32 mov eax, [READ_DATA_SIZE]
 	a32 mov [INITRD_SIZE], eax
 	mov al, '!'
@@ -1385,7 +1385,7 @@
 	dw 0
 
 KERNEL_CMD_LINE:
-	db "debug loglevel=7 earlycon=uart8250,io,0x3f8,9600n8 console=tty0 console=ttyS0,9600n8 rdinit=/bin/sinit root=/dev/ram0 rootfstype=ramfs iommu=off", 0
+	db "debug loglevel=7 earlyprintk=vga earlycon=uart8250,io,0x3f8,9600n8 console=tty0 console=ttyS0,9600n8 init=/init iommu=off vga=0x311", 0
 
 KERNEL_CMD_SIZE equ $-KERNEL_CMD_LINE
 
