diff -rauN uflbbl/src/boot.asm uflbbl-kernel-params-patch/src/boot.asm
--- uflbbl/src/boot.asm	2023-05-05 18:48:22.000000000 +0200
+++ uflbbl-kernel-params-patch/src/boot.asm	2023-05-07 19:51:21.735659678 +0200
@@ -1385,7 +1385,7 @@
 	dw 0
 
 KERNEL_CMD_LINE:
-	db "debug loglevel=7 earlycon=uart8250,io,0x3f8,9600n8 console=tty0 console=ttyS0,9600n8 rdinit=/bin/sinit root=/dev/ram0 rootfstype=ramfs iommu=off", 0
+	db "debug loglevel=7 earlycon=uart8250,io,0x3f8,9600n8 console=tty0 console=ttyS0,9600n8 init=/init iommu=off nomodeset nokaslr", 0
 
 KERNEL_CMD_SIZE equ $-KERNEL_CMD_LINE
 
