diff -rauN a/configure b/configure
--- a/configure	2026-03-04 19:38:20.000000000 +0100
+++ b/configure	2026-03-04 19:43:22.445630267 +0100
@@ -195,8 +195,8 @@
 test "$i" -gt 50 && fail "$0: cannot create temporary file $tmpc"
 done
 set +C
-trap 'rm -f "$tmpc" "$tmpo"' EXIT QUIT TERM HUP
-trap 'rm -f "$tmpc" "$tmpo" && echo && fail "$0: interrupted"' INT
+#trap 'rm -f "$tmpc" "$tmpo"' EXIT QUIT TERM HUP
+#trap 'rm -f "$tmpc" "$tmpo" && echo && fail "$0: interrupted"' INT
 
 #
 # Find a C compiler to use
@@ -302,7 +302,7 @@
 : ${PKG_CONFIG:=pkg-config}
 have_pkgconfig=no
 printf "checking for $PKG_CONFIG... "
-cmdexists $PKG_CONFIG && have_pkgconfig=yes
+#cmdexists $PKG_CONFIG && have_pkgconfig=yes
 printf "%s\n" "$have_pkgconfig"
 
 if test "$curses" != "no" ; then
@@ -334,7 +334,7 @@
 		fi
 
 		CFLAGS_CURSES=""
-		LDFLAGS_CURSES="-l$libcurses"
+		LDFLAGS_CURSES="-l$libcurses -ltermcap"
 
 		if $CC $CFLAGS $CFLAGS_CURSES "$tmpc" \
 			$LDFLAGS $LDFLAGS_CURSES -o "$tmpo" >/dev/null 2>&1 ; then
