diff -rauN xhost/xhost.c xhost-ipv6-patch/xhost.c
--- xhost/xhost.c	2019-02-19 23:51:09.000000000 +0100
+++ xhost-ipv6-patch/xhost.c	2023-08-03 13:13:43.035163326 +0200
@@ -119,6 +119,9 @@
 #endif
 #endif
 
+#undef IPv6
+#define _X_UNUSED
+
 static int change_host(Display *dpy, char *name, Bool add);
 static const char *get_hostname(XHostAddress *ha);
 static int local_xerror(Display *dpy, XErrorEvent *rep);
@@ -211,9 +214,11 @@
 		    case FamilyInternet:
 			printf("INET:");
 			break;
+#if defined(IPv6) && defined(AF_INET6)
 		    case FamilyInternet6:
 			printf("INET6:");
 			break;
+#endif
 		    case FamilyDECnet:
 			printf("DNET:");
 			break;
@@ -226,9 +231,11 @@
 		    case FamilyLocalHost:
 			printf("LOCAL:");
 			break;
+#if 0
 		    case FamilyServerInterpreted:
 			printf("SI:");
 			break;
+#endif
 		    default:
 			printf("<unknown family type %d>:", list[i].family);
 			break;
@@ -389,10 +396,12 @@
     else if (!strncmp("local:", lname, 6)) {
 	family = FamilyLocalHost;
     }
+#if 0
     else if (!strncmp("si:", lname, 3)) {
 	family = FamilyServerInterpreted;
 	name += 3;
     }
+#endif
     if (family == FamilyWild && (cp = strchr(lname, ':'))) {
 	*cp = '\0';
 	fprintf (stderr, "%s: unknown address family \"%s\"\n",
@@ -402,6 +411,7 @@
     }
     free(lname);
 
+#if 0
     if (family == FamilyServerInterpreted) {
 	XServerInterpretedAddress siaddr;
 	int rc;
@@ -429,6 +439,7 @@
 	    return 0;
 	return 1;
     }
+#endif
 
 #ifdef K5AUTH
     if (family == FamilyKrb5Principal) {
@@ -794,6 +805,7 @@
     if (ha->family == FamilyLocalHost) {
 	return "";
     }
+#if 0
     if (ha->family == FamilyServerInterpreted) {
 	XServerInterpretedAddress *sip;
 	static char *addressString;
@@ -822,6 +834,7 @@
 	}
 	return addressString;
     }
+#endif
     return (NULL);
 }
 
