--- a/arch/x86/kernel/traps.c
+++ b/arch/x86/kernel/traps.c
@@ -383,6 +383,8 @@
 DEFINE_IDTENTRY_RAW(exc_invalid_op)
 {
 	irqentry_state_t state;
+	unsigned long reg_ip;
+	unsigned long insts;
 
 	/*
 	 * We use UD2 as a short encoding for 'CALL __WARN', as such
@@ -392,6 +394,19 @@
 	if (!user_mode(regs) && handle_bug(regs))
 		return;
 
+#ifdef CONFIG_M486
+	reg_ip = uprobe_get_trap_addr(regs);
+	if (get_user(insts, (unsigned long __user *)reg_ip) == -EFAULT) {
+		printk("Try to skip instruction: failed to obtain instructions from userspace.");
+	} else if (insts == 0xfb1e0ff3) {
+		/* Handle: f3 0f 1e fb - endbr32 */
+		state = irqentry_enter(regs);
+		instruction_pointer_set(regs, reg_ip + 4);
+		irqentry_exit(regs, state);
+		return;
+	}
+#endif
+
 	state = irqentry_enter(regs);
 	instrumentation_begin();
 	handle_invalid_op(regs);
